pkg/server/events.go

```diff
@@ -42,6 +42,13 @@ const (
 	backOffInitDuration        = 1 * time.Second
 	backOffMaxDuration         = 5 * time.Minute
 	backOffExpireCheckDuration = 1 * time.Second
+
+	// handleEventTimeout is the timeout for handling 1 event. Event monitor
+	// handles events in serial, if one event blocks the event monitor, no
+	// other events can be handled.
+	// Add a timeout for each event handling, timeout events will be requeued and
+	// handled again in the future.
+	handleEventTimeout = 5 * time.Second
```

Comment:
why 5 seconds? Is that a standard convention? This seems short, would not want to cause an endless loop.

Reply:
Not a convention. I just want to keep it as short as possible, because it will block the whole event monitor.
We don't do much when handling an event, I think 5 second should be good enough.